posthog/hogql_queries/legacy_compatibility/feature_flag.py

```diff
@@ -1,26 +1,44 @@
-from typing import cast
 import posthoganalytics
 from django.conf import settings
 from posthog.cloud_utils import is_cloud
 from posthog.models.user import User
-from django.contrib.auth.models import AnonymousUser
 
+from posthog.schema import InsightType
 
-def hogql_insights_enabled(user: User | AnonymousUser) -> bool:
+
+GLOBAL_FLAG = "hogql-insights-preview"
+INSIGHT_TYPE_TO_FLAG: dict[InsightType, str] = {
+    InsightType.TRENDS: "hogql-insight-trends",
+    InsightType.FUNNELS: "hogql-insights-funnels",
+    InsightType.RETENTION: "hogql-insights-retention",
+    InsightType.PATHS: "hogql-insights-paths",
+    InsightType.LIFECYCLE: "hogql-insights-lifecycle",
+    InsightType.STICKINESS: "hogql-insights-stickiness",
+}
+
+
+def hogql_insights_enabled(user: User, insight_type: InsightType) -> bool:
     if settings.HOGQL_INSIGHTS_OVERRIDE is not None:
         return settings.HOGQL_INSIGHTS_OVERRIDE
 
-    # on PostHog Cloud, use the feature flag
-    if is_cloud():
-        if not hasattr(user, "distinct_id"):  # exclude api endpoints that don't have auth from the flag
-            return False
-
-        return posthoganalytics.feature_enabled(
-            "hogql-insights",
-            cast(str, user.distinct_id),
-            person_properties={"email": user.email},
-            only_evaluate_locally=True,
-            send_feature_flag_events=False,
-        )
-    else:
+    if not is_cloud():
+        # HogQL insights are only available on Cloud
```

Comment:
```suggestion
        # HogQL insights are only available on Cloud until public release
```

Reply:
Actually this whole `if` was kind of pointless, since there's the flag - removed it